src/main/java/ay2021s1_cs2103_w16_3/finesse/logic/commands/EditCommand.java

```diff
@@ -105,7 +107,13 @@ private static Transaction createEditedTransaction(Transaction transactionToEdit
         Set<Category> updatedCategories = editTransactionDescriptor.getCategories()
                 .orElse(transactionToEdit.getCategories());
 
-        return new Transaction(updatedTitle, updatedAmount, updatedDate, updatedCategories);
+        if (transactionToEdit instanceof Expense) {
+            return new Expense(updatedTitle, updatedAmount, updatedDate, updatedCategories);
+        } else if (transactionToEdit instanceof Income) {
+            return new Income(updatedTitle, updatedAmount, updatedDate, updatedCategories);
+        } else {
+            return new Transaction(updatedTitle, updatedAmount, updatedDate, updatedCategories);
+        }
```

Comment:
This is quite bad design. Perhaps we should just throw an error when executing any of the base commands which should never be executed?

Reply:
Agree. This will be handled in a future PR.

Reply:
Refer to issue #125.